kafka-connect-common/src/main/scala/io/lenses/streamreactor/connect/cloud/common/sink/commit/CommitContext.scala

```diff
@@ -15,20 +15,44 @@
  */
 package io.lenses.streamreactor.connect.cloud.common.sink.commit
 
+import com.typesafe.scalalogging.LazyLogging
 import io.lenses.streamreactor.connect.cloud.common.model.TopicPartitionOffset
 
+trait CommitContext {
+  def count:                Long
+  def fileSize:             Long
+  def createdTimestamp:     Long
+  def lastFlushedTimestamp: Option[Long]
+
+  def lastModified: Long = lastFlushedTimestamp.getOrElse(createdTimestamp)
+
+  def logFlush(flushing: Boolean, result: Seq[ConditionCommitResult]): Unit
+}
+
 /**
   * @param tpo              the [[TopicPartitionOffset]] of the last record written
   * @param count            the number of records written thus far to the file
   * @param createdTimestamp the time in milliseconds when the the file was created/accessed first time
   */
-case class CommitContext(
+case class CloudCommitContext(
   tpo:                  TopicPartitionOffset,
   count:                Long,
   fileSize:             Long,
   createdTimestamp:     Long,
   lastFlushedTimestamp: Option[Long],
   partitionFile:        String,
-) {
-  def lastModified: Long = lastFlushedTimestamp.getOrElse(createdTimestamp)
+) extends CommitContext
+    with LazyLogging {
+  override def logFlush(flushing: Boolean, result: Seq[ConditionCommitResult]): Unit = {
```

Comment:
I wouldn't embed the logging call in the `CommitContext`. Generating the string would be fine but it feels to me that the representation of the commit context (data only) and a log line are two distinct concerns. 

One issue with embedding here the log line is that the line will say it was produced in the `CloudCommitContext` . It would likely be more useful it it mention that it was produced either in `HttpWriter` or `Writer`

Reply:
If it's not clear what I mean is that this method should only return the String and then it would be up to the Writer and/or HttpWriter to log it

Reply:
I definitely understand what you mean.  This would be better.

The reason it is done like this, is to make the most of the typesafe logger framework's macro support.

Simply returning a string, If debug is disabled, then all the evaluation and string concatenation etc will occur but will be wasted.

I wonder if I could follow your approach but instead of returning a String, return a () => String so as to avoid the unnecessary processing?

Reply:
Indeed,  good point ! 

Having a look at the macro itself however I believe it might actually be ok, provided you don't instantiate the string outside the log call.

```scala
 q"if ($underlying.isDebugEnabled) $underlying.debug($message, _root_.scala.Array[AnyRef](${anyRefArgs.head}, ${anyRefArgs(1)}): _*)"
 ```
 
 Seems like it just rewrites the line as: 
 ```scala
 if(logger.isDebugEnabled){
    logger.debug(...)
 }
 ```
 
 so the string won't be allocated unless debug is enabled